plugins/dates/src/api/parse/range/02-date-range.js

```diff
@@ -138,4 +139,25 @@ export default [
       return null
     },
   },
+
+  {
+    // 2 to 4 weeks
+    match: '[<min>#Value] to [<max>#Value] [<unit>#Duration]',
+    desc: '2 to 4 weeks',
+    parse: (m, context) => {
+      const { min, max, unit } = m.groups()
+
+      let start = new Unit(context.today, null, context)
+      let end = start.clone()
+
+      const duration = unit.text('implicit')
+      start = start.applyShift({ [duration]: min.numbers().get()[0] })
+      end = end.applyShift({ [duration]: max.numbers().get()[0] })
+
+      return {
+        start: start,
+        end: end,
```

Comment:
Need to sort a conditional where if the duration phrase is `day`, `month`, or `year` then subtract a day and call `end.end()`. 

```js
      end = end.applyShift({ [duration]: max.numbers().get()[0], day: -1 })

      return {
        start: start,
        end: end.end(),
```

Reply:
oh, right - hmm.
There's different types of `Unit` there if you wanted to use them - the idea was that they each call `.end()` or `.middle()` and it means something different for a Month or an Hour. Not sure it worked out v. well, of if that helps in your case.
Worse you could grab the spacetime object with `Unit.d` (sorry!) and add a condition there
cheers

Reply:
Thanks @spencermountain, I'll give it a try but may just target bigger units then like day/week/month/year for now. The plugin is called `compromise-dates` after all, not `compromise-times`.

Should i throw tests for this in a existing file (if so which?), or do you think this needs a new one?

Reply:
sure, up to you!